ush/python/pygfs/task/atm_analysis.py

```diff
@@ -0,0 +1,405 @@
+        local_dict = AttrDict(
+            {
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
+                'npz_ges': self.config.LEVS - 1,
+                'npz': self.config.LEVS - 1,
+                'npx_anl': _res_anl + 1,
+                'npy_anl': _res_anl + 1,
+                'npz_anl': self.config.LEVS - 1,
+                'ATM_WINDOW_BEGIN': _window_begin,
+                'ATM_WINDOW_LENGTH': f"PT{self.config.assim_freq}H",
+                'comin_ges_atm': self.config.COMIN_GES,
+                'OPREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'APREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'GPREFIX': f"gdas.t{self.runtime_config.previous_cycle.hour:02d}z.",
+                'fv3jedi_yaml': _fv3jedi_yaml,
+            }
+        )
+
+        # task_config is everything that this task should need
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self: Analysis) -> None:
+        """Initialize a global atm analysis
+
+        This method will initialize a global atm analysis using JEDI.
+        This includes:
+        - staging CRTM fix files
+        - staging FV3-JEDI fix files
+        - staging B error files
+        - staging model backgrounds
+        - generating a YAML file for the JEDI executable
+        - linking the JEDI executable (TODO make it copyable, requires JEDI fix)
+        - creating output directories
+        """
+        super().initialize()
+
+        # stage CRTM fix files
+        crtm_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_crtm_coeff.yaml')
+        logger.debug(f"Staging CRTM fix files from {crtm_fix_list_path}")
+        crtm_fix_list = parse_yamltmpl(crtm_fix_list_path, self.task_config)
+        FileHandler(crtm_fix_list).sync()
+
+        # stage fix files
+        jedi_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_jedi_fix.yaml')
+        logger.debug(f"Staging JEDI fix files from {jedi_fix_list_path}")
+        jedi_fix_list = parse_yamltmpl(jedi_fix_list_path, self.task_config)
+        FileHandler(jedi_fix_list).sync()
+
+        # stage berror files
+        # copy static background error files, otherwise it will assume ID matrix
+        if self.task_config.STATICB_TYPE in ['identity', 'bump', 'gsibec']:
+            logger.debug(f"Stage files for STATICB_TYPE {self.task_config.STATICB_TYPE}")
+            FileHandler(self.get_berror_dict(self.task_config)).sync()
+        else:
+            raise WorkflowException(f"***ERROR*** STATICB_TYPE = {self.task_config.STATICB_TYPE} is invalid")
+
+        # stage backgrounds
+        FileHandler(self.get_bkg_dict(AttrDict(self.task_config, **self.task_config))).sync()
```

Comment:
What is happening here?
Why are two instances of `self.task_config` being merged?

Shouldn't this just be:
```
        FileHandler(self.get_bkg_dict(self.task_config)).sync()
```

Reply:
@aerorahul ,  `atm_analysis.py` originated as a copy of `aero_analysis.py`.   `aero_analysis.py` has
```
        # stage backgrounds
        FileHandler(self.get_bkg_dict(AttrDict(self.task_config, **self.task_config))).sync()
```
I blindly copied this into `atm_analysis.py`.

I simplified the line in a working copy of `atm_analysis.py` to your recommended change.   It works.   Shall I commit this change to my `feature/refactor_jedivar` or would you prefer to add it to your `patch/pr1372`?

Reply:
Yeah. Please update in your branch. 
It seems aero_analysis.py should also be updated.

Reply:
Change committed at [8492cc](https://github.com/NOAA-EMC/global-workflow/pull/1372/commits/8492ccf72a856c4c9328885742540ae37cc1b916).   This commit also includes your changes from patch PR [#3](https://github.com/RussTreadon-NOAA/global-workflow/pull/3)

Reply:
May this conversation be resolved?